source/fab/repository.py

```diff
@@ -38,6 +39,38 @@ def extract(self, target: Path):
         raise NotImplementedError("Abstract methods must be implemented.")
 
 
+class GitRepo(Repository):
+    """
+    Extracts a source tree from a Git repository.
+
+    This class currently wraps the Git command-line client but that may prove
+    unsatisfactory. In particular it could be too slow. In that case we should
+    examine the possibility of using Python bindings.
+    """
+    _TIMEOUT = 4  # seconds
+
+    def __init__(self, url: str):
+        super().__init__(url)
+
+    def extract(self, target: Path):
+        target.parent.mkdir(parents=True, exist_ok=True)
+        command = ['git', 'archive',
+                   '--format', 'tar',
+                   '--remote', self.url,
+                   'HEAD']
+        process = Popen(command, stdout=PIPE)
+
+        extractor = tarfile.open(fileobj=process.stdout, mode='r|')
+        extractor.extractall(target)
+
+        process.wait(self._TIMEOUT)
+        if process.returncode != 0:
+            message = "Fault exporting tree from Git repository."
+            if process.stderr:
+                message += '\n' + str(process.stderr)
```

Comment:
I'm not sure that `process.stderr` will contain anything unless you pass `stderr=PIPE` to `Popen` (maybe I'm wrong though?)  Also does the message work correctly here or would it need to be something like `process.stderr.decode()`?)

Reply:
The documentation is unclear on this subject so I tried it out and it looks like you are right.